UI/MapboxNavigationUI/RouteMapViewController.swift

```diff
@@ -0,0 +1,427 @@
+                controller?.distanceLabel.text = nil
+                controller?.streetLabel.text = upComingStep.instructions
+            } else {
+                controller?.distanceLabel.text = distanceFormatter.string(from: distanceRemaining)
+                
+                if let name = upComingStep.names?.first ?? destinations, let ref = upComingStep.codes?.first {
+                    let foregroundColor = NavigationUI.shared.secondaryTextColor
+                    let attributes = [
+                        NSForegroundColorAttributeName: foregroundColor,
+                        NSFontAttributeName: controller?.streetLabel.font,
+                        ]
+                    let attributedString = NSMutableAttributedString(string: "\(name) ", attributes: attributes)
+                    let attachment = NSTextAttachment()
+                    attributedString.append(NSAttributedString(attachment: attachment))
+                    
+                    if controller?.streetLabel.attributedText?.string != attributedString.string {
+                        controller?.streetLabel.attributedText = attributedString
+                        
+                        let components = ref.components(separatedBy: " ")
+                        if components.count > 1 {
+                            shieldAPIDataTask = dataTaskForShieldImage(network: components[0], number: components[1], height: 32 * UIScreen.main.scale) { (image) in
+                                controller?.streetLabel.attributedText = nil
+                                if let image = image {
+                                    attachment.bounds = CGRect(x: 0, y: 0, width: image.size.width / UIScreen.main.scale, height: image.size.height / UIScreen.main.scale)
+                                    attachment.image = image
+                                }
+                                controller?.streetLabel.attributedText = attributedString
+                            }
+                            shieldAPIDataTask?.resume()
+                        }
+                    }
+                } else {
+                    controller?.streetLabel.text = upComingStep.names?.first ?? destinations
+                }
+            }
+        }
+        
+        controller?.turnArrowView.step = routeProgress.currentLegProgress.upComingStep
+    }
+    
+    func dataTaskForShieldImage(network: String, number: String, height: CGFloat, completion: @escaping (UIImage?) -> Void) -> URLSessionDataTask? {
+        guard let imageNamePattern = ShieldImageNamesByPrefix[network] else {
+            return nil
+        }
+        
+        let imageName = imageNamePattern.replacingOccurrences(of: " ", with: "_").replacingOccurrences(of: "{ref}", with: number)
+        let apiURL = URL(string: "https://commons.wikimedia.org/w/api.php?action=query&format=json&maxage=86400&prop=imageinfo&titles=File%3A\(imageName)&iiprop=url%7Csize&iiurlheight=\(Int(round(height)))")!
```

Comment:
This is part of the public Wikimedia Commons API. The `maxage` parameter should keep repeated requests from being expensive, but we should ensure that we’re calling the API only as frequently as necessary.